src/fileopener/plugin.ts

```diff
@@ -92,19 +176,18 @@ class FileOpener implements IFileOpener {
       return this._open(handlers[0], path);
 
     // If there were no matches, look for default handler(s).
-    } else if (handlers.length === 0) {
-      for (let h of this._handlers) {
-        if (isDefaultProperty.get(h)) handlers.push(h);
-      }
+    } else if (handlers.length === 0 && this._defaultHandler !== null) {
+       return this._open(this._defaultHandler, path);
     }
 
     // If there we no matches, do nothing.
     if (handlers.length == 0) {
-      return Promise.reject(new Error(`Could not open file '${path}'`));
+      throw new Error(`Could not open file '${path}'`);
 
     // If there was one handler, use it.
     } else if (handlers.length === 1) {
       return this._open(handlers[0], path);
```

Comment:
I think if there is no default handler registered, we shouldn't default to the only handler.